File: cpp/questions/majorityElement.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int v = 0;
        int s;
        for (const auto n: nums) {
            if (v == 0) {
                s = n;
                ++v;
            } else if (s == n) {
                ++v;
            } else {
                --v;
            }
        }
        return s;
    }
};

File: typescript/questions/majorityElement.ts
function majorityElement(nums: number[]): number {
  let v = 0;
  let p = 0;
  for (const n of nums) {
    if (v == 0) {
      p = n;
      v = 1;
    } else {
      v += n == p ? 1 : -1;
    }
  }
  return p;
}
